packages/editor/src/components/copy-handler/index.js

```diff
@@ -63,14 +63,20 @@ class CopyHandler extends Component {
 export default compose( [
 	withSelect( ( select ) => {
 		const {
-			getMultiSelectedBlocks,
 			getMultiSelectedBlockClientIds,
-			getSelectedBlock,
+			getSelectedBlockClientId,
+			getBlocks,
 		} = select( 'core/editor' );
+
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const selectedBlockClientIds = selectedBlockClientId ? [ selectedBlockClientId ] : getMultiSelectedBlockClientIds();
+
 		return {
-			multiSelectedBlocks: getMultiSelectedBlocks(),
-			multiSelectedBlockClientIds: getMultiSelectedBlockClientIds(),
-			selectedBlock: getSelectedBlock(),
+			selectedBlockClientIds,
+
+			// We only care about this value when the copy is performed
+			// We call it dynamically in the event handler to avoid unnecessary re-renders.
+			getBlocks,
```

Comment:
I've used the same technique used in the other PR (cc @gziolo @aduth), it's becoming a pattern, we might want to find a more elegant solution in a follow-up PR. (`withDispatch` access to `select` somehow)

Reply:
As far as I remember `withDispatch` wraps those handlers with proxies to ensure they don’t update on every prop change. So you can’t execute `select` outside of those handlers as they will become immediately stale. We will have to document it properly to avoid any bugs. It is advanced usage which is necessary mostly in core, so it might be okey to attack it this way.

Reply:
I will open PR and let's discuss it there.